src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -570,12 +570,18 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
 
     // TODO This won't remove validations for ledgers that do not get
     // validated. That will likely require inserting LedgerSeq into
-    // the validations table
+    // the validations table.
+    //
+    // This query has poor performance with large data sets.
+    // The schema needs to be redesigned to avoid the JOIN, or an
+    // RDBMS that supports concurrency should be used.
+    /*
     clearSql (*ledgerDb_, lastRotated,
         "SELECT MIN(LedgerSeq) FROM Ledgers;",
         "DELETE FROM Validations WHERE LedgerHash IN "
         "(SELECT Ledgers.LedgerHash FROM Validations JOIN Ledgers ON "
         "Validations.LedgerHash=Ledgers.LedgerHash WHERE Ledgers.LedgerSeq < %u);");
+     */
```

Comment:
Looking at the old query, I wonder if how simply removing the join would affect performance.

eg:
```
        DELETE FROM Validations WHERE LedgerHash IN 
       (SELECT Ledgers.LedgerHash FROM Ledgers 
        WHERE Ledgers.LedgerSeq < %u);
```
The subquery will probably return many more ledger hashes this way, but depending on how the `Validations` table is indexed, how SQLLite handles the extra info, etc, it may still be faster.